### PR TITLE
set DAGSTER_RUN_JOB_NAME env var

### PR DESCRIPTION
**Pull request by alangenfeld**
similar to the labels, add `DAGSTER_RUN_` env vars to environment for containerized executors and launchers 

### How I Tested These Changes

added test, can add some more

### CODE REVIEW

**Review by gibsondan — DISMISSED** [review]
should we do this in the other launchers and executors as well to make it consistent?

File: python_modules/libraries/dagster-celery-k8s/dagster_celery_k8s/executor.py [hunk 1]
```diff
@@ -369,6 +369,13 @@ def _execute_step_k8s_job(
                 "dagster/op": step_key,
                 "dagster/run-id": execute_step_args.pipeline_run_id,
             },
+            env_vars=[
+                {
+                    "name": "DAGSTER_RUN_JOB_NAME",
+                    "value": pipeline_run.pipeline_name,
```
Comment (gibsondan) [inline python_modules/libraries/dagster-celery-k8s/dagster_celery_k8s/executor.py:375]: job_name?

**Review by gibsondan — APPROVED** [review]

File: python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/launcher_tests/test_launching.py [hunk 2]
```diff
@@ -148,8 +147,9 @@ def test_launcher_dont_use_current_task(
     assert container_definition["image"] == image
     assert not container_definition.get("entryPoint")
     assert not container_definition.get("dependsOn")
-    # It takes in the environment configured on the instance
-    assert container_definition["environment"] == environment
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-aws/dagster_aws_tests/ecs_tests/launcher_tests/test_launching.py:152]: is this asserting against important behavior? All i did was append the new env var and it seems to change this inheritance scheme?

File: python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py [hunk 3]
```diff
@@ -435,6 +435,8 @@ def _run_task_kwargs(self, run, image, container_context) -> Dict[str, Any]:
         definition if needed.
         """
         environment = self._environment(container_context)
+        environment.append({"name": "DAGSTER_RUN_JOB_NAME", "value": run.job_name})
```
Comment (alangenfeld) [inline python_modules/libraries/dagster-aws/dagster_aws/ecs/launcher.py:438]: [1]

**Pull request merged**